apps/common/src/main/scala/org/lfdecentralizedtrust/splice/store/UpdateHistory.scala

```diff
@@ -2173,11 +2163,15 @@ object UpdateHistory {
   )
 
   case class State(
-      historyId: Option[Long]
+      historyId: Option[Long],
+      corruptSnapshotsDeleted: Boolean,
```

Comment:
We could also remove this flag and have `ScanHistoryBackfillingTrigger` independently check for corrupt snapshots, at the cost of calling `migrationsWithCorruptSnapshots` more often.

`migrationsWithCorruptSnapshots` executes 3 independent SQL queries, each of them taking <10ms to run on CILR.